### PR TITLE
port content from frost-form-container

### PR DESCRIPTION
**Pull request by rox163**
#major#

### CODE REVIEW

**Comment by rox163** [comment]
@sglanzer addon transition rules Ok?

File: addon/transitions.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+export default function () {
+  this.transition(
+    this.inHelper('liquid-modal'),
+    this.toModal(/(?:edit-|create-)/),
```
Comment (sglanzer) [inline addon/transitions.js]: So these would be `edit-user`, `create-user`, I think I'd prefer this to follow more pod based thinking where the category leads, followed by specifics - e.g. `user-*`, which would switch the edit and create to a suffix.  Sound ok?

File: addon/transitions.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    this.use('explode', {
+      pick: '.lf-overlay',
+      use: ['cross-fade', {
+        maxOpacity: 0.1,
+        duration: 200
+      }]
+    }, {
+      pick: '.lm-container',
+      use: ['toDown']
```
Comment (sglanzer) [inline addon/transitions.js]: Explicit duration?  I assume this default to 250ms

File: tests/dummy/app/pods/components/frost-create-form/component.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import Ember from 'ember'
+import layout from './template'
+
+export default Ember.Component.extend({
+  layout,
+  modalForms: Ember.inject.service('modal-forms'),
+  bunsenModel: {
```
Comment (sglanzer) [inline tests/dummy/app/pods/components/frost-create-form/component.js]: Let's not call it bunsen when we use it - this is a `userAccountModel` or something like that

File: tests/dummy/app/pods/components/frost-create-form/component.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      }
+    },
+    'required': [
+      'username', 'password'
+    ]
+  },
+
+  bunsenView: {},
+
+  willInsertElement () {
```
Comment (sglanzer) [inline tests/dummy/app/pods/components/frost-create-form/component.js]: Can you raise an issue to handle this more cleanly in the future?  I'd like the style of the overlay to not need this type of service to activate

File: tests/dummy/app/pods/components/frost-create-form/template.hbs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{{#frost-modal-input title='Test title' subtitle='Subtitle'}}
+{{!#frost-form-container titleComponent='myTitle'}}
+
+  {{frost-bunsen-form
```
Comment (sglanzer) [inline tests/dummy/app/pods/components/frost-create-form/template.hbs]: I don't like that we call it `bunsen` in the component (I mean, how is that significant to anyone consuming this) - maybe we make a transition to just `frost-form` in the future

File: tests/dummy/app/pods/components/frost-create-form/template.hbs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+      model=bunsenModel
+      view=bunsenView
+      onChange=(action 'formValueChanged')
```
Comment (sglanzer) [inline tests/dummy/app/pods/components/frost-create-form/template.hbs]: We need to be consistent with our event names.  Currently we're using dasherized (e.g. on-change)

File: tests/dummy/app/pods/components/frost-create-form/template.hbs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+      onValidation=(action 'onValidation')}}
+
+  <div class='actions'>
```
Comment (sglanzer) [inline tests/dummy/app/pods/components/frost-create-form/template.hbs]: Can we default the actions?

**Comment by sglanzer** [comment]
Just need the change from prefix to suffix for the `-edit` / `-create`, the rest can be address afterward

File: tests/dummy/app/pods/components/frost-create-form/template.hbs (hunk 6 above)
Comment (rox163) [inline tests/dummy/app/pods/components/frost-create-form/template.hbs]: requires a change in ember-frost-bunsen. I'll raise an issue there.

File: tests/dummy/app/pods/components/frost-create-form/template.hbs (hunk 5 above)
Comment (rox163) [inline tests/dummy/app/pods/components/frost-create-form/template.hbs]: created an issue in ember-frost-bunsen

**Comment by sglanzer** [comment]
Approved

**Pull request merged**